### PR TITLE
use beeline 2.0

### PR DESCRIPTION
**Pull request by lizthegrey**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Description
Uses [beeline v2.0](https://github.com/honeycombio/beeline-ruby/releases/tag/v2.0.0) to benefit from improved arg parsing.

## Related Tickets & Documents
#6512

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
n/a

## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

### CODE REVIEW

**Comment by lizthegrey** [comment]
Passing this sidekiq failure to @paulosman:

```
  1) Sidekiq::HoneycombMiddleware sends an event with expected keys
     Failure/Error: collected_data = Honeycomb.client.send("client").events.map(&:data).detect { |h| h["sidekiq.args"] == expected_hash["sidekiq.args"] }
     
     NoMethodError:
       undefined method `client' for #<Honeycomb::Client:0x0000000008573020>
     # ./spec/lib/sidekiq/honeycomb_middleware_spec.rb:34:in `block (2 levels) in <top (required)>'
     # ./spec/rails_helper.rb:102:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:102:in `block (2 levels) in <top (required)>'
  2) Sidekiq::HoneycombMiddleware without args sends an event with expected keys
     Failure/Error: collected_data = Honeycomb.client.send("client").events.map(&:data).detect { |h| h["sidekiq.args"] == expected_hash["sidekiq.args"] }
     
     NoMethodError:
       undefined method `client' for #<Honeycomb::Client:0x0000000008573020>
     # ./spec/lib/sidekiq/honeycomb_middleware_spec.rb:54:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:102:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:102:in `block (2 levels) in <top (required)>'
  3) Sidekiq::HoneycombMiddleware with an error sends an event with expected keys
     Failure/Error: collected_data = Honeycomb.client.send("client").events.map(&:data).detect { |h| h["sidekiq.args"] == error_hash["sidekiq.args"] }
     
     NoMethodError:
       undefined method `client' for #<Honeycomb::Client:0x0000000008573020>
     # ./spec/lib/sidekiq/honeycomb_middleware_spec.rb:76:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:102:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:102:in `block (2 levels) in <top (required)>'
```

**Comment by lizthegrey** [comment]
aha, `spec/lib/sidekiq/honeycomb_middleware_spec.rb` shouldn't exist, that test is already tested inside of the beeline tests. going to add to this pull to delete the extra spurious test.

File: spec/lib/sidekiq/honeycomb_middleware_spec.rb [hunk 1]
```diff
@@ -31,7 +31,7 @@ def perform(arg = nil)
       TestSidekiqWorker.perform_async("dont fail")
     end
 
-    collected_data = Honeycomb.client.send("client").events.map(&:data).detect { |h| h["sidekiq.args"] == expected_hash["sidekiq.args"] }
+    collected_data = Honeycomb.libhoney.events.map(&:data).detect { |h| h["sidekiq.args"] == expected_hash["sidekiq.args"] }
```
Comment (mstruve) [inline spec/lib/sidekiq/honeycomb_middleware_spec.rb]: This works! i just ran these specs locally with the upgrade and change :D

**Comment by lizthegrey** [comment]
This worked amazingly.
Screenshot from 2020-03-10 13-27-09
Screenshot from 2020-03-10 13-26-55

**Pull request merged**